File: sdk/v1.0.58/javascript/oauth/src/http.js
'use strict';

const { OAuthError } = require('./errors');

/**
 * @param {string} url
 * @param {Record<string, string>} params
 * @param {Record<string, string>} [headers]
 * @returns {Promise<unknown>}
 */
async function postForm(url, params, headers = {}) {
  const body = new URLSearchParams(params).toString();
  const response = await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/x-www-form-urlencoded', ...headers },
    body,
  });
  const json = await response.json();
  if (!response.ok) throw OAuthError.fromResponse(response.status, json);
  return json;
}

/**
 * @param {string} url
 * @param {Record<string, string>} [headers]
 * @returns {Promise<unknown>}
 */
async function getJson(url, headers = {}) {
  const response = await fetch(url, { headers });
  const json = await response.json();
  if (!response.ok) throw OAuthError.fromResponse(response.status, json);
  return json;
}

/**
 * @param {string} url
 * @param {object} body
 * @param {Record<string, string>} [headers]
 * @returns {Promise<unknown>}
 */
async function postJson(url, body, headers = {}) {
  const response = await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json', ...headers },
    body: JSON.stringify(body),
  });
  const json = await response.json();
  if (!response.ok) throw OAuthError.fromResponse(response.status, json);
  return json;
}

module.exports = { postForm, getJson, postJson };

File: sdk/archive/v1.0.81/python/oauth_sdk/_http.py
from __future__ import annotations

import httpx

from ._errors import OAuthError


def post_form(
    url: str,
    params: dict[str, str],
    client: httpx.Client,
    headers: dict[str, str] | None = None,
) -> dict:
    response = client.post(url, data=params, headers=headers or {})
    body = response.json()
    if not response.is_success:
        raise OAuthError.from_response(response.status_code, body)
    return body


def post_json(
    url: str,
    body: dict,
    client: httpx.Client,
    headers: dict[str, str] | None = None,
) -> dict:
    response = client.post(url, json=body, headers=headers or {})
    data = response.json()
    if not response.is_success:
        raise OAuthError.from_response(response.status_code, data)
    return data


def get_json(
    url: str,
    client: httpx.Client,
    headers: dict[str, str] | None = None,
) -> dict:
    response = client.get(url, headers=headers or {})
    body = response.json()
    if not response.is_success:
        raise OAuthError.from_response(response.status_code, body)
    return body
